ios-target/View/Home/MatchNotificationViewController.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  MatchNotificationViewController.swift
+//  ios-target
+//
+//  Created by sol manini on 7/3/19.
+//  Copyright © 2019 TopTier labs. All rights reserved.
+//
+
+import UIKit
+
+class MatchNotificationViewController: ModalViewController {
+  
+  @IBOutlet weak var connectMatchButton: UIButton!
+  @IBOutlet weak var userNameLabel: UILabel!
+  @IBOutlet weak var userImage: UIImageView!
+  
+  var viewModel = MatchNotificationViewModel()
+  weak var delegate: MatchNotificationViewModelDelegate?
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    connectMatchButton.backgroundColor = .macaroniAndCheese
+    if let userEmail = viewModel.user?.email {
+      userNameLabel.text = userEmail
+    }
+    setImage()
+    
+  }
+  
+  func setImage() {
+    if let imageURL = viewModel.user?.image {
+      userImage.setImage(with: imageURL, placeholder: "userPlaceholder")
+    } else {
+      userImage.image = UIImage.random()
```

Comment:
I don't think a random image is a good placeholder, perhaps your `setImage` function could receive an optional URL and set the placeholder in case it is nil. So you won't need to do this kind of checks every time

Reply:
mm i dont think I understand what you mean

Reply:
since `setImage` already accepts an optional url . you can do:
`userImage.setImage(with: viewModel.user?.image, placeholder: "userPlaceholder")`
and if the url is nil it will use the placeholder and not a random image. makes sense ?

Reply:
got it! thanks!